### PR TITLE
Tweak settings for hover

### PR DESCRIPTION
**Pull request by nabobalis**
Hopefully should address some of the issues we have with it.

### CODE REVIEW

**Comment by nabobalis** [comment]
I want to check its behaviour on RTD first.

**Comment by Cadair** [comment]
Does this work on the preview builds, it seems to not be working for me?

**Comment by nabobalis** [comment]
TODO:

See if it's possible to disable the pop up fully for refs or URLs.

File: docs/conf.py [hunk 1]
```diff
@@ -212,12 +212,18 @@
...
 
-hoverxref_tooltip_maxwidth = 600  # RTD main window is 696px
 hoverxref_auto_ref = True
-hoverxref_mathjax = True
-# hoverxref has to be applied to these
 hoverxref_domains = ["py"]
-
+hoverxref_mathjax = True
+hoverxref_modal_hover_delay = 500
+hoverxref_tooltip_maxwidth = 600  # RTD main window is 696px
```
Comment (nabobalis) [inline docs/conf.py:219]: The default is 450, if we want to go narrower.

File: docs/conf.py [hunk 2]
```diff
@@ -212,14 +212,34 @@
         # A RTD thing to avoid a CSRF block when docs are using a custom domain
         hoverxref_api_host = "/_"
 
-hoverxref_tooltip_maxwidth = 600  # RTD main window is 696px
-hoverxref_auto_ref = True
-hoverxref_mathjax = True
-# hoverxref has to be applied to these
+hoverxref_auto_ref = False
```
Comment (nabobalis) [inline docs/conf.py:215]: I had to disable this, I can't get it to not work on gallery examples.

Either I am missing a config value or we need to consider a patch upstream.

Comment (nabobalis) [reply docs/conf.py:215]: I think if we could change: https://github.com/readthedocs/sphinx-hoverxref/blob/b4b5cf4ae95cbb78151509ed46b5b0c5f821ec58/hoverxref/domains.py#L60 to string match instead, we could add the generic gallery text.

**Comment by Cadair** [comment]
I want to see this built before approving, especially as it's hard/impossible to make hover stuff work locally.

File: docs/conf.py [hunk 3]
```diff
@@ -212,14 +212,34 @@
...
 hoverxref_domains = ["py"]
-
+hoverxref_mathjax = True
+hoverxref_modal_hover_delay = 500
+hoverxref_tooltip_maxwidth = 600  # RTD main window is 696px
+hoverxref_intersphinx = [
```
Comment (Cadair) [inline docs/conf.py:220]: Is this just a list of all our intersphinx mappings or have you curated it?

File: docs/conf.py [hunk 4]
```diff
@@ -212,14 +212,34 @@
...
+    "astropy",
+    "matplotlib",
```
Comment (Cadair) [inline docs/conf.py:222]: does mpl work as it's not on rtd?

File: docs/conf.py [hunk 5]
```diff
@@ -212,14 +212,34 @@
...
+    "numpy",
+    "python",
+    "scipy",
+    "hvpy",
+    "asdf",
+    "zeep",
```
Comment (Cadair) [inline docs/conf.py:228]: Do we really need this?

File: docs/conf.py [hunk 6]
```diff
@@ -212,14 +212,34 @@
...
+    "sunkit_instruments",
+    "sunkit_image",
+    "sqlalchemy",
```
Comment (Cadair) [inline docs/conf.py:231]: and this

File: docs/conf.py (hunk 3 above)
Comment (nabobalis) [reply docs/conf.py:220]: The first one.

File: docs/conf.py [hunk 7]
```diff
@@ -229,13 +249,12 @@
     "meth": "tooltip",
     "mod": "tooltip",
     "obj": "tooltip",
-    # roles with std domain
+    # Roles within the standard domain
```
Comment (Cadair) [inline docs/conf.py:252]: the std domain is a *thing*

```suggestion
    # Roles within the std domain
```

File: docs/conf.py (hunk 4 above)
Comment (nabobalis) [reply docs/conf.py:222]: They special case several of them. Scipy, Numpy and MPL should all work as they support that manually.

File: docs/conf.py (hunk 5 above)
Comment (nabobalis) [reply docs/conf.py:228]: Why not? Surely we want this to work for links that are API and we link to zeep classes.

File: docs/conf.py (hunk 7 above)
Comment (nabobalis) [reply docs/conf.py:252]: I will accept after RTD builds.

File: docs/conf.py (hunk 3 above)
Comment (Cadair) [reply docs/conf.py:220]: If we are going to do that then we can just do `list(intersphinx_mapping.keys())` or whatever, but should we curate it?

Comment (nabobalis) [reply docs/conf.py:220]: Fair, I can do that instead. 

I personally think if we intersphinx, we would want them all to hover when we link to those classes.

**Review by Cadair — APPROVED** [review]

**Pull request merged**